va/va.c

```diff
@@ -1432,18 +1432,52 @@ VAStatus vaMapBuffer (
 )
 {
   VADriverContextP ctx;
-  VAStatus va_status;
+  VAStatus va_status = VA_STATUS_SUCCESS;
   
   CHECK_DISPLAY(dpy);
   ctx = CTX(dpy);
-  
+
   VA_FOOL_FUNC(va_FoolMapBuffer, dpy, buf_id, pbuf);
-  
-  va_status = ctx->vtable->vaMapBuffer( ctx, buf_id, pbuf );
+  if (ctx->vtable->vaMapBuffer2)
```

Comment:
It's really a good optimization because of the flags can save the un-needed copy.
The misc usage of vtable->vaMapBuffer and vtable->vaMapBuffer2 within vaMapBuffer vaMapBuffer2 APIs may be verbose. For vaMapBuffer2 API, if driver does not provide the virtual function point, we can return VA_STATUS_ERROR_UNIMPLEMENTED directly. The app user should consider the fallback cases.

Reply:
in the future, I suppose, somedriver will just implement vaMapBuffer2, to be compatible with old app, need vaMapBuffer also be supported. this why I add similar fallback logic in vaMapBuffer and vaMapBuffer2